homeassistant/components/control4/light.py

```diff
@@ -73,40 +73,71 @@ async def async_update_data_dimmer():
     await dimmer_coordinator.async_refresh()
 
     items_of_category = await get_items_of_category(hass, entry, CONTROL4_CATEGORY)
+
+    entity_list = []
     for item in items_of_category:
-        if item["type"] == CONTROL4_ENTITY_TYPE:
-            item_name = item["name"]
-            item_id = item["id"]
-            item_parent_id = item["parentId"]
-            item_is_dimmer = item["capabilities"]["dimmer"]
-
-            if item_is_dimmer:
-                item_coordinator = dimmer_coordinator
+        try:
+            if item["type"] == CONTROL4_ENTITY_TYPE:
+                item_name = item["name"]
+                item_id = item["id"]
+                item_parent_id = item["parentId"]
+
+                item_manufacturer = None
+                item_device_name = None
+                item_model = None
+
+                for parent_item in items_of_category:
+                    if parent_item["id"] == item_parent_id:
+                        item_manufacturer = parent_item["manufacturer"]
+                        item_device_name = parent_item["name"]
+                        item_model = parent_item["model"]
```

Comment:
Is there a specific one that will be missing, or can any key be unexpectedly be missing?  If it is only one, `"key" in item` or `"key" in parent_item` would be cleaner than wrapping the whole block an a try.   

If all of them can throw, disregard.

Reply:
I want to be able to catch a missing key in any of the variables, because that would be a deviation from the known API response format, which I would like to log for investigation.